badges in playerstats

ChaosMarc opened the pull request:
![badges](https://cloud.githubusercontent.com/assets/1343751/18817635/e4d29186-8365-11e6-9b67-2d3128367d08.PNG)

ChaosMarc committed:
badges in playerstats

Cptspal commented:
Great !
I've just one error in main.js line 430, shouldn't it be 
`current_value = self.user_data[self.settings.users[user_id].username].pokedex[20].inventory_item_data.pokedex_entry.times_captured;`
(the .username is missing)
At least, it's better for me like this. Thx.

ChaosMarc committed:
fixed pikachu_caught lookup

ChaosMarc committed:
fixed settings lookup

ChaosMarc commented:
hey Cptspal,

you're right! I was testing my code in an older version of the web module where the lookup worked without `.username`. This is now fixed.

While fixing that I recognized another error with the pikachu badge. My previous lookup took for granted that the array of pokedex_entries was always the same and used index[20]... I added a new lookup to get pikachu all the time and not the pokemon at index[20].

please recheck my latest commit :)

Cptspal reviewed (approved):
Nice PR !

mmixx commented on js/main.js:
```diff
@@ -389,7 +393,7 @@ var mapView = {
       case 1:
         var current_user_stats = self.user_data[self.settings.users[user_id].username].stats[0].inventory_item_data.player_stats;
-        $('#subtitle').html('Trainer Info');
+        $('#subtitle').html(self.settings.users[user_id]);
```
Can you add the `.username` on line 396?

```
$('#subtitle').html(self.settings.users[user_id]);
```

So it will fetch the username used... and also, on my previously committed code... I've added

```
$("div").find("[data-bot-id='" + self.settings.users[user_id].username + "']").html()
```

Every time the `Info` button is clicked, it will fetch the name of the bot that is currently set (when the bot runs `UpdateLiveStats`) on the top of the bot menus (not the username)...

ChaosMarc replied:
I readded your changes to the username display. I must have overlooked this line while merging. 
Additionally I observed that 'pokemon_caught_by_type' is not present for freshly created users and my pikachu lookup in the pokedex failed if it hadn't been caught yet. I fixed those to errors, too.

mmixx commented:
Nice PR... so we can easily monitor which badge we completed even if we're not logged in using our phone :)

ChaosMarc committed:
fixed username display

ChaosMarc committed:
added existence checks for caught_by_type and pikachu_caught

ChaosMarc committed:
readded lost code due to merge conflicts

mmixx reviewed (approved):
Nice...
